### ISSUE TITLE
Recognition from files not possible

### ISSUE THREAD

**Issue by JamesDean829** [marin-m/SongRec#34]
TLDR: File input does not work as intended, only microphone recognition.

Hey, 

SongRec is quite well in recognizing files via the desktop microphone. 

However most of the songs I am tagging on the go on my smartphone (Android). I was really happy that there is an alternative to the shazam app. When I tried to record audio files on my phone and copying them onto the laptop to SongRec, it does not recognize them. 

The audio files is always displayed as "Unrecognized format" and the .wav-file gives an empty json string (only some metadata), but no song. However when I play the song the microphone recognition puts out the right song (so it does not depend on a bad quality of the .wav-record).

I don't know why the file input does not work with both file formats, I even tried the command line but with the same result :( I can upload these files if you want.

PS *Dreaming*: It would be really cool, if this app will ever find its way on androids (f-droid).

Kind regards

**Comment by marin-m** [comment]
Hello,

Can you share the files that you have tried to use with the application so that we can check what the format actually is (in the case where there would be a mismatch, etc.)?

You can upload the files directory through clicking "Attach file by..." on the bottom of the Github comment text area, and selecting these on your dick.

Alternatively, could you type the `file NAME_OF_YOUR_FILE.wav` command in a terminal, and share the output?

Thank you

**Comment by JamesDean829** [comment]
Hey, 

here are the two files :) The .wav has been recorded by the native recorder of Lineage OS and the .mp3 by the [Simple Voice Recorder](https://f-droid.org/en/packages/com.simplemobiletools.voicerecorder/). 

Kind regards
Julian

[Audio-Records.zip](https://github.com/marin-m/SongRec/files/6217212/Audio-Records.zip)

**Comment by marin-m** [comment]
Hello,

Thanks for sharing. It seems that the ".mp3" file as actually a renamed MP4 (AAC) file. SongRec is currently limited by the use of the underlying [`rodio`](https://github.com/RustAudio/rodio) library which only supports actual .MP3, .WAV, .FLAC, .OGG files. It would be likely better to fall back on a FFMpeg-based solution to support more formats (see issue #9), but back when I last looked I didn't find a functional and easily integrable solution.

Regarding the non-recognition of the WAV file, it seems to be due to the behavior of recognizing only at mots the central 12 seconds of your file when you try to recognize a file (rather than a sliding window of 12 seconds updated every 4 seconds when you are recognizing with the microphone).

**Comment by JamesDean829** [comment]
Hey, 
thank you for your reply. I tried it with another audio recorder app, which is more reliable by claiming to record in mp3 and this file worked.

And it will be be cool if the app will support a sliding window in the future in order to detect file inputs longer than 12 seconds :)
